app/controllers/posts_controller.rb

```diff
@@ -22,9 +22,24 @@ def index
     @posts = current_user.followed_posts.ordered_posts
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @post.likes.destroy_all
+    delete_comments_and_likes(@post.comments)
```

Comment:
You can add dependent: :destroy to both has_many associations in post.rb, so there is no need to define the delete_comments_and_likes method.